src/main/java/org/tb/dailyreport/rest/WorkingDayRestEndpoint.java

```diff
@@ -0,0 +1,117 @@
+        var employeecontract = employeecontractDAO.getEmployeeContractByEmployeeIdAndDate(authorizedUser.getEmployeeId(), date);
+        if(employeecontract == null) {
+            throw new ResponseStatusException(NOT_FOUND);
+        }
+
+        try {
+            var wd = ofNullable(workingdayDAO.getWorkingdayByDateAndEmployeeContractId(date, employeecontract.getId())).orElseGet(Workingday::new);
+            wd.setEmployeecontract(employeecontract);
+            wd.setRefday(date);
+            wd.setStarttimehour(data.getStarthour());
+            wd.setStarttimeminute(data.getStartminute());
+            wd.setBreakhours(data.getBreakhours());
+            wd.setBreakminutes(data.getBreakminutes());
+            workingdayDAO.save(wd);
+        } catch (AuthorizationException e) {
+            throw new ResponseStatusException(UNAUTHORIZED, "Could not create workingday. " + e.getErrorCode());
+        } catch (InvalidDataException e) {
+            throw new ResponseStatusException(BAD_REQUEST, "Could not create workingday. " + e.getErrorCode());
+        }
+    }
+
+    @GetMapping(produces = APPLICATION_JSON_VALUE)
+    @ResponseStatus(OK)
+    @Operation
+    public WorkingDayData getToday() {
+        return doGet(DateUtils.today());
+    }
+
+    @GetMapping(path = "/{date}", produces = APPLICATION_JSON_VALUE)
+    @ResponseStatus(OK)
+    @Operation
+    public WorkingDayData get(@PathVariable String date) {
+        return doGet(DateUtils.parse(date));
+    }
+
+    private WorkingDayData doGet(LocalDate date){
+        if (!authorizedUser.isAuthenticated()) {
+            throw new ResponseStatusException(UNAUTHORIZED);
+        }
+
+        var employeecontract = employeecontractDAO.getEmployeeContractByEmployeeIdAndDate(authorizedUser.getEmployeeId(), date);
+        if(employeecontract == null) {
+            throw new ResponseStatusException(NOT_FOUND);
+        }
+
+        var workingDay = workingdayDAO.getWorkingdayByDateAndEmployeeContractId(date, employeecontract.getId());
+        if(workingDay == null) {
+            throw new ResponseStatusException(NOT_FOUND);
+        }
+
+        return mapToWorkingDayData(workingDay);
+    }
+
+    private WorkingDayData mapToWorkingDayData(Workingday wd) {
+        return WorkingDayData.builder()
+                .id(wd.getId())
+                .starthour(wd.getStarttimehour())
+                .startminute(wd.getStarttimeminute())
+                .breakhours(wd.getBreakhours())
+                .breakminutes(wd.getBreakminutes())
+                .date(DateUtils.format(wd.getRefday()))
+                .build();
+    }
+
+    @DeleteMapping(consumes = APPLICATION_JSON_VALUE)
+    @ResponseStatus(OK)
+    @Operation
+    public void delete(@RequestBody WorkingDayData data) {
+        if (!authorizedUser.isAuthenticated()) {
+            throw new ResponseStatusException(UNAUTHORIZED);
+        }
+
+        if (data.getId() != -1) {
```

Comment:
check is missing if user is owner of working day
discuss: only the id should be send as a parameter?

Reply:
discuss: or even better provide a date a the parameter?

Reply:
switched to use date parameter